homeassistant/components/stream/__init__.py

```diff
@@ -92,6 +117,18 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     hass.data[DOMAIN] = {}
     hass.data[DOMAIN][ATTR_ENDPOINTS] = {}
     hass.data[DOMAIN][ATTR_STREAMS] = []
+    if (conf := config.get(DOMAIN)) and conf[CONF_LL_HLS]:
+        StreamConstants.LL_HLS = True
+        assert isinstance(conf[CONF_SEGMENT_DURATION], float)
+        assert isinstance(conf[CONF_PART_DURATION], float)
```

Comment:
Instead of asserts here, it's better to define `TypedDict` for `conf`.

Reply:
Then we have to call `cast(StreamConf,conf)`?

Reply:
I prefer not using cast unless necessary.
```python
conf: StreamConfig = config.get(DOMAIN)
```
You'll also need to declare constants as `Final` to use them as keys of `TypedDict`.

Reply:
I see. Anyway, creating a TypeDict is an extra hoop to jump through to avoid the asserts.
I guess we should circle back and see what the actual problem with the asserts is. I'm guessing there are 2 distinct issues with it (please correct me if I am wrong):
1) Using assert where an exception should be used instead - an assert should not be checking something which can fail in production code
2) There is a performance penalty for assert
For the cases in this PR, I don't think 1) is a concern. I think 2) is only a concern if the optimize flag is not passed to the interpreter. @allenporter Do you know where you saw the advice re: asserts?

Reply:
I think i was wrong about asserts, apologies.